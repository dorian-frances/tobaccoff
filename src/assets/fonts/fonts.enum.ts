export enum FontsEnum {
  MEDIUM = 'Montserrat-Medium',
  SEMI_BOLD = 'Montserrat-SemiBold',
  BOLD = 'Montserrat-Bold',
  EXTRA_BOLD = 'Montserrat-ExtraBold',
}
