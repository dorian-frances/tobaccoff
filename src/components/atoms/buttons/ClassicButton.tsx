import React from 'react';
import { Button } from 'react-native-paper';
import { ColorsEnum } from '../../../assets/colors/colors.enum';
import { StyleProp, TextStyle } from 'react-native';

type ButtonComponentProps = {
  text: string;
  labelStyle: StyleProp<TextStyle>;
  mode: 'elevated' | 'outlined';
  onPress: () => void;
  buttonColor?: string;
};

const ClassicButton = ({
  text = 'Lorem Ipsum',
  labelStyle,
  mode = 'elevated',
  buttonColor = ColorsEnum.PRIMARY_COLOR_BUTTON,
  onPress = () => console.log('Button pressed !'),
}: ButtonComponentProps) => {
  return (
    <Button
      mode={mode}
      onPress={onPress}
      labelStyle={labelStyle}
      loading={false}
      buttonColor={buttonColor}
      contentStyle={{ paddingTop: 5, height: 45 }}
    >
      {text}
    </Button>
  );
};

export default ClassicButton;
