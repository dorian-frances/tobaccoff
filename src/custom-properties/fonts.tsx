const customFonts = {
  regular: {
    fontFamily: 'Montserrat-Regular',
  },
  medium: {
    fontFamily: 'Montserrat-Medium',
  },
  semiBold: {
    fontFamily: 'Montserrat-SemiBold',
  },
  bold: {
    fontFamily: 'Montserrat-Bold',
  },
  extraBold: {
    fontFamily: 'Montserrat-ExtraBold',
  },
};

export default customFonts;
