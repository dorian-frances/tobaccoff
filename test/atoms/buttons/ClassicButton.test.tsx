import { render } from '@testing-library/react-native';
import ClassicButton from '../../../src/components/atoms/buttons/ClassicButton';
import { FontsEnum } from '../../../src/assets/fonts/fonts.enum';
import React from 'react';

describe('<ClassicButton/>', () => {
  const fakeButtonText = 'fakeButtonText';
  const fakeMode = 'elevated';
  const labelStyle = {
    fontFamily: FontsEnum.BOLD,
    color: 'black',
    fontSize: 20,
  };
  const fakeOnPress = jest.fn();

  it('should render correctly', () => {
    const classicButton = render(
      <ClassicButton
        labelStyle={labelStyle}
        text={fakeButtonText}
        mode={fakeMode}
        onPress={fakeOnPress}
      />
    );
    expect(classicButton.toJSON()).toMatchSnapshot();
  });

  it('should access button text', () => {
    const classicButton = render(
      <ClassicButton
        text={fakeButtonText}
        labelStyle={labelStyle}
        mode={fakeMode}
        onPress={fakeOnPress}
      />
    );
    const foundText = classicButton.getByText(fakeButtonText);
    expect(foundText).toBeTruthy();
  });
});
