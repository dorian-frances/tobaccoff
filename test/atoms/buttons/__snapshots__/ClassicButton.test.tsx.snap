// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ClassicButton/> should render correctly 1`] = `
<View
  collapsable={false}
  style={
    {
      "alignSelf": undefined,
      "bottom": undefined,
      "end": undefined,
      "flex": undefined,
      "left": undefined,
      "position": undefined,
      "right": undefined,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.15,
      "shadowRadius": 3,
      "start": undefined,
      "top": undefined,
    }
  }
  testID="button-container-outer-layer"
>
  <View
    collapsable={false}
    style={
      {
        "flex": undefined,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 1,
      }
    }
    testID="button-container-inner-layer"
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "#E8DEF8",
          "borderColor": "transparent",
          "borderRadius": 20,
          "borderStyle": "solid",
          "borderWidth": 0,
          "flex": undefined,
          "minWidth": 64,
        }
      }
      testID="button-container"
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            false,
            {
              "borderRadius": 20,
            },
          ]
        }
        testID="button"
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              {
                "height": 45,
                "paddingTop": 5,
              },
            ]
          }
        >
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              [
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 20,
                },
                {
                  "textAlign": "left",
                },
                {
                  "writingDirection": "ltr",
                },
                [
                  {
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  false,
                  {
                    "marginHorizontal": 24,
                    "marginVertical": 10,
                  },
                  undefined,
                  false,
                  {
                    "color": "rgba(103, 80, 164, 1)",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "letterSpacing": 0.1,
                    "lineHeight": 20,
                  },
                  undefined,
                ],
              ]
            }
            testID="button-text"
          >
            fakeButtonText
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
